Source/CsDebugScript.Engine/Engine/Debuggers/DbgEngDllHelpers/DebuggeFlowController.cs

```diff
@@ -0,0 +1,111 @@
+﻿using DbgEngManaged;
+using System;
+
+namespace CsDebugScript.Engine.Debuggers.DbgEngDllHelpers
+{
+    /// <summary>
+    /// Controler for Debugee actions during live debugging.
+    /// </summary>
+    class DebuggeeFlowController
+    {
+        /// <summary>
+        /// Signal fired during interactive process debugging when debugee is released.
+        /// </summary>
+        public System.Threading.AutoResetEvent DebugStatusGo { get; private set; }
+
+        /// <summary>
+        /// Signal fired during interactive process debugging when debugee is interrupted.
+        /// </summary>
+        public System.Threading.AutoResetEvent DebugStatusBreak { get; private set; }
+
+        /// <summary>
+        /// Loop responsible for catching debug events and signaling debugee state.
+        /// </summary>
+        private System.Threading.Thread debuggerStateLoop;
+
+        /// <summary>
+        /// Syncronization signaling that debug callbacks are installed.
+        /// </summary>
+        private static readonly object eventCallbacksReady = new Object();
+
+        /// <summary>
+        /// Debug client for gbgeng interaction.
+        /// </summary>
+        private static IDebugClient client;
+
+        /// <summary>
+        /// Constructor.
+        /// </summary>
+        /// <param name="originalClient">IDebugClient</param>
+        public DebuggeeFlowController(IDebugClient originalClient)
+        {
+            DebugStatusGo = new System.Threading.AutoResetEvent(false);
+
+            // Default is that we start in break mode.
+            // Needs to be changed when we allow non intrusive attach/start for example.
+            //
+            DebugStatusBreak = new System.Threading.AutoResetEvent(true);
+
+            client = originalClient.CreateClient();
```

Comment:
Seems like I don't need it. Although I have to admit I don't fully understand when I need to create one.

---
In reply to: [69982125](https://github.com/southpolenator/WinDbgCs/pull/8#discussion_r69982125) [](ancestors = 69982125)